### PR TITLE
[bitnami/keycloak] automounting of serviceAccountToken

### PR DESCRIPTION
**Pull request by KuroObi**
It enables controll over the automounting of the service account token in the pod.
By disabling the automount, potential
attackers cannot access the Kubernetes API on behalf/through the pod.

This commit enables it by default, but leaves it configurable if anyone
needs to use it.

**Description of the change**

This change gives a user finer control over security aspects if it is needed.

**Benefits**

Disabling the automounting reduces an attack vector inside the k8s cluster against the kube-api.

**Possible drawbacks**

An inexperienced user could select a keycloak clustering method that relies on contact with the kube-api, which would be not possible without a mounted AccountToken.

**Checklist** 
- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [X] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. [bitnami/<name_of_the_chart>])

### CODE REVIEW

File: bitnami/keycloak/templates/statefulset.yaml [hunk 1]
```diff
@@ -41,6 +41,7 @@ spec:
         {{- end }}
     spec:
       serviceAccountName: {{ template "keycloak.serviceAccountName" . }}
+      automountServiceAccountToken: {{ .Values.automountServiceAccountToken }}
```
Comment (migruiz4) [inline bitnami/keycloak/templates/statefulset.yaml:44]: The value used is wrong value, it should be `.Values.serviceAccount.automountServiceAccountToken`
```suggestion
      automountServiceAccountToken: {{ .Values.serviceAccount.automountServiceAccountToken }}
```

File: bitnami/keycloak/values.yaml [hunk 2]
```diff
@@ -788,6 +788,9 @@ serviceAccount:
   ## If not set and create is true, a name is generated using the fullname template
   ##
   name: ""
+  ## @param serviceAccount.automountServiceAccountToken Auto-mount the service account token in the pod
+  ##
+  automountServiceAccountToken: true
```
Comment (migruiz4) [inline bitnami/keycloak/values.yaml:793]: Please set the default value to `false`.

As you mentioned in your description, it is safer to not automount the service account token, and we are trying to follow a 'secure by default' approach.

>By disabling the automount, potential attackers cannot access the Kubernetes API on behalf/through the pod.

**Review by migruiz4 — CHANGES_REQUESTED** [review]
Thank you very much for your contribution!

In addition to the below comments, please be aware that you are adding `automountServiceAccountToken` to the Keycloak Statefulset where it will only apply to the keycloak pod. 
I think it would make sense to add it at `serviceaccount.yaml` instead.

File: bitnami/keycloak/templates/statefulset.yaml (hunk 1 above)
Comment (KuroObi) [reply bitnami/keycloak/templates/statefulset.yaml:44]: I've added the option to dis-/en-able it in the pod and the ServiceAccount.
This adds the possibility to disable the service Account and also disable automounting the default Service Account token.

**Comment by migruiz4** [comment]
Hi @KuroObi,

Thank you for applying the proposed changes but I would like to ask you to drop the value `automountServiceAccountToken` (pod level) from your PR, for this chart to be aligned with the other charts that use `serviceAccount.automountServiceAccountToken`.

We are aware that setting `automountServiceAccountToken` granularly for each pod would be a different feature, but our charts do not support it yet and we may decide on a different approach for it in the future.

**Comment by KuroObi** [comment]
> Hi @KuroObi,
> 
> Thank you for applying the proposed changes but I would like to ask you to drop the value `automountServiceAccountToken` (pod level) from your PR, for this chart to be aligned with the other charts that use `serviceAccount.automountServiceAccountToken`.
> 
> We are aware that setting `automountServiceAccountToken` granularly for each pod would be a different feature, but our charts do not support it yet and we may decide on a different approach for it in the future.

I have removed the pod level adjustability and aligned the PR to the other Charts as requested.

**Review by migruiz4 — APPROVED** [review]
Thank you for your contribution! Looks good to me!

**Pull request merged**